base/pkg/dir.jl

```diff
@@ -37,7 +37,24 @@ function init(meta::AbstractString=DEFAULT_META, branch::AbstractString=META_BRA
     metadata_dir = joinpath(dir, "METADATA")
     if isdir(metadata_dir)
         info("Package directory $dir is already initialized.")
-        LibGit2.set_remote_url(metadata_dir, meta)
+        with(LibGit2.GitRepo, metadata_dir) do r
+            LibGit2.transact(r) do repo
+                LibGit2.set_remote_url(repo, meta)
+
+                # Ideally this logic would be contained within LibGit2
+                ref = LibGit2.lookup_branch(repo, branch, true)  # remote branch
+                isnull(ref) && (ref = LibGit2.lookup_branch(repo, branch, false))  # local branch
```

Comment:
Any ideas about fixing #17512? If `meta` is an invalid URL, this will cause it to fallback on the local branch.